index.js

```diff
@@ -1,64 +1,68 @@
 'use strict';
 
-var assign = require('object.assign');
+const { declare } = require('@babel/helper-plugin-utils');
 
-var modules = [require('babel-plugin-transform-es2015-modules-commonjs'), {
-  strictMode: true, // add "use strict"
-  strict: false // this allows __esModule to be exported
-}];
-
-var defaultTargets = {
+const defaultTargets = {
   android: 30,
   chrome: 35,
   edge: 14,
   explorer: 9,
   firefox: 52,
   safari: 8,
-  ucandroid: 1
+  ucandroid: 1,
 };
 
-function buildTargets(options) {
-  return assign({}, defaultTargets, options.additionalTargets);
+function buildTargets({ additionalTargets }) {
+  return Object.assign({}, defaultTargets, additionalTargets);
 }
 
-module.exports = function buildAirbnbPreset(context, options) {
-  var transpileTargets = (options && options.targets) || buildTargets(options || {});
+module.exports = declare((api, options) => {
+  api.assertVersion(7);
+
+  const debug = typeof options.debug === 'boolean' ? options.debug : false;
 
-  var debug = (options && typeof options.debug === 'boolean') ? !!options.debug : false;
+  const {
+    targets = buildTargets(options),
+    devEnv = 'development',
+    modules,
+    removePropTypes,
+  } = options;
 
   return {
     presets: [
-      require('babel-preset-env').default(null, {
-        debug: debug,
+      [require('@babel/preset-env'), {
+        debug,
         exclude: [
           'transform-async-to-generator',
-          'transform-es2015-template-literals',
-          'transform-regenerator'
+          'transform-template-literals',
+          'transform-regenerator',
         ],
-        modules: false,
-        targets: transpileTargets
-      }),
-      require('babel-preset-react')
+        modules,
+        targets,
+      }],
+      [require('@babel/preset-react'), { development: api.env(devEnv) }],
```

Comment:
I agree with you that this may be easier as a boolean exposed for the user to pass, since it's not clear that the semantics are worth hard-coding into this preset. @ljharb Since this wasn't a feature present in Babel 6 anyway, is hard-coding this preset to check `NODE_ENV` actually desirable? If so, it would be helpful to spell out exactly your expectations. What is configurable by the user and what are we comparing it against?

Reply:
@loganfsmyth I would expect that NODE_ENV is the primary mechanism to control whether the react preset is in dev mode or production mode; also having an explicit option seems useful.

So in other words, I'd expect the config to be 1) react dev env option, or 2) NODE_ENV

Reply:
Should the preset option be a boolean then, with the default being set based on `NODE_ENV ==="development"`?